src/lib/node-oauth-client-provider.ts

```diff
@@ -168,6 +209,10 @@ export class NodeOAuthClientProvider implements OAuthClientProvider {
       authorizationUrl.searchParams.set('resource', this.authorizeResource)
     }
 
+    const effectiveScope = this.getEffectiveScope()
+    authorizationUrl.searchParams.set('scope', effectiveScope)
+    debugLog('Added scope parameter to authorization URL', { scopes: effectiveScope })
```

Comment:
Hello! I'm running into this issue myself and really appreciate this PR. Unfortunately when using your branch, I'm still not seeing the correct scopes attached. This piece of code successfully attaches the scope for me, but I'm not sure how to integrate it into your `getEffectiveScope` method. 

```suggestion
    const effectiveScope = this.getEffectiveScope()
    authorizationUrl.searchParams.set('scope', effectiveScope)
    debugLog('Added scope parameter to authorization URL', { scopes: effectiveScope })
    
    // The correct custom scopes are found in the authorization URL resource + using the discoverOAuthProtectedResourceMetadata
    const resourceURL = authorizationUrl.searchParams.get('resource')
    if (resourceURL) {
      const metadata = await discoverOAuthProtectedResourceMetadata(resourceURL)
      if (metadata.scopes_supported !== undefined) {
        authorizationUrl.searchParams.set('scope', metadata.scopes_supported.join(' '))
      }
    }
```

Any guidance on how to integrate this into the PR would be very much appreciated! Hoping to get this fixed to address [this reported issue](https://github.com/geelen/mcp-remote/issues/162).